### PR TITLE
Enable formating with pyink to adhere with google3 style. 

### PR DESCRIPTION
**Pull request by Obliviour**
## Fixes / Features
- [Enable formating with pyink to adhere with google3 style.](https://github.com/google/xpk/commit/37ae3cce23beba58c147808ba559a9f740985a69)
- [Add proper pre commit check and documentation.](https://github.com/google/xpk/commit/d16898726edfee3147c0fdcd3eb07a9811c0c9a3)

## Testing / Documentation
- [ y] Tests pass
- [ y] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -25,8 +25,78 @@ Guidelines](https://opensource.google/conduct/).
...
+# 1. Code will be autoformatted on:
+git commit -m "My cool new feature"
+# 2. If code needs to be reformatted, check what changes were made, and add them to the commit.
+git add -p
+# 3. Run git commit again, which should successfully pass pre-commit checks.
+git commit -m "My cool new feature"
+
+
+# Manually format code by running:
+plink .
```
Comment (RoshaniN) [inline docs/contributing.md:73]: Perhaps a typo?

File: docs/contributing.md [hunk 2]
```diff
@@ -25,8 +25,78 @@ Guidelines](https://opensource.google/conduct/).
...
+pip3 install pyink==24.3.0
+```
+
+To lint the code:
+
+```sh
+# Format files in the local directory.
+pyink .
+# Check if files need to be formatted.
+pylink --check .
```
Comment (RoshaniN) [inline docs/contributing.md:101]: Did you mean pyink?

File: pylintrc [hunk 3]
```diff
@@ -215,7 +212,7 @@ no-docstring-rgx=(__.*__|main|test.*|.*test|.*Test)$
 
 # Minimum line length for functions/classes that require docstrings, shorter
 # ones are exempt.
-docstring-min-length=10
+docstring-min-length=12
```
Comment (RoshaniN) [inline pylintrc:215]: Are these pyink recommended values?

**Review by RoshaniN — CHANGES_REQUESTED** [review]
Thanks Victor!

File: pylintrc (hunk 3 above)
Comment (Obliviour) [reply pylintrc:215]: These are recommended values from https://google.github.io/styleguide/pyguide.html

File: pylintrc [hunk 4]
```diff
@@ -401,12 +401,3 @@ valid-classmethod-first-arg=cls,
 
 # List of valid names for the first argument in a metaclass class method.
 valid-metaclass-classmethod-first-arg=mcs
-
-
-[EXCEPTIONS]
-
-# Exceptions that will emit a warning when being caught. Defaults to
-# "Exception"
-overgeneral-exceptions=builtins.StandardError,
```
Comment (RoshaniN) [inline pylintrc:410]: Curious to know why this section was removed!

File: docs/contributing.md [hunk 5]
```diff
@@ -25,8 +25,78 @@ Guidelines](https://opensource.google/conduct/).
...
+internal xpk development from a cloudtop machine.
+
+```shell
+  # Create venv if needed.
+  ## One time step of creating the venv
+  VENV_DIR=~/venvp3
+  python3 -m venv $VENV_DIR
+  ## enter the venv.
+  source $VENV_DIR/bin/activate
+  ## Same steps as above.
```
Comment (SurbhiJainUSC) [inline docs/contributing.md:42]: What do you mean by `Same steps as above` here? Maybe this is not required.

File: docs/contributing.md [hunk 6]
```diff
@@ -25,8 +25,78 @@ Guidelines](https://opensource.google/conduct/).
...
+  git clone https://github.com/google/xpk.git
```
Comment (SurbhiJainUSC) [inline docs/contributing.md:43]: Also, I think this is already covered in Step 1, so can be removed from here.

File: docs/contributing.md [hunk 7]
```diff
@@ -25,8 +25,78 @@ Guidelines](https://opensource.google/conduct/).
...
+# 1. Code will be autoformatted on:
+git commit -m "My cool new feature"
+# 2. If code needs to be reformatted, check what changes were made, and add them to the commit.
+git add -p
+# 3. Run git commit again, which should successfully pass pre-commit checks.
+git commit -m "My cool new feature"
+
+
+# Manually format code by running:
+pyink .
```
Comment (SurbhiJainUSC) [inline docs/contributing.md:73]: What about pylint command? Is that not required?

File: docs/contributing.md [hunk 8]
```diff
@@ -25,8 +25,78 @@ Guidelines](https://opensource.google/conduct/).
...
+1. Install developer tools including `pyink`, `pylint`, and `precommit` using
+
+```shell
+  git clone https://github.com/google/xpk.git
+  pip install .[dev]
+```
+
+2. Install git hook scripts.
+```shell
+cd $XPK_REPO_DIR
```
Comment (RoshaniN) [inline docs/contributing.md:56]: This needs to be specified above.

File: pylintrc [hunk 9]
```diff
@@ -72,26 +76,28 @@ disable=apply-builtin,
...
         long-builtin,
         long-suffix,
         map-builtin-not-iterating,
+        misplaced-comparison-constant,
+        missing-function-docstring,
         metaclass-assignment,
         next-method-called,
         next-method-defined,
         no-absolute-import,
-        no-else-break,
```
Comment (RoshaniN) [inline pylintrc:90]: This may still be useful?

File: docs/contributing.md (hunk 7 above)
Comment (Obliviour) [reply docs/contributing.md:73]: This shows how to manually format if needed. The pre-commit check will autoformat for you when you run git commit ...

**Review by RoshaniN — APPROVED** [review]

**Pull request merged**